src/contact.rs

```diff
@@ -11,64 +11,60 @@ use crate::population_loader::Alive;
-        -> Option<PersonId>;
+pub trait ContextContactExt {
+    /// Returns a potential contact for the transmitter.
+    /// Returns Ok(None) if there are no eligible contacts.
+    /// In the future, this function can be expanded to return a
+    /// contact specific to the person's household or to weight
+    /// drawing contacts be a contact matrix.
+    ///
+    /// Errors
+    /// - If there is only one person in the population.
+    fn get_contact(&mut self, transmitter_id: PersonId) -> Result<Option<PersonId>, IxaError>;
 }
 
-impl QueryContacts for Context {
-    fn get_contact(&mut self, transmitee_id: PersonId) -> Result<Option<PersonId>, IxaError> {
+#[allow(clippy::cast_possible_truncation)]
+#[allow(clippy::cast_sign_loss)]
+fn sample_person_from_list(context: &mut Context, list: &[PersonId], weights: &[f64]) -> PersonId {
+    let index = context.sample_distr(ContactRng, Categorical::new(weights).unwrap());
+    list[index as usize]
+}
+
+impl ContextContactExt for Context {
+    fn get_contact(&mut self, transmitter_id: PersonId) -> Result<Option<PersonId>, IxaError> {
         if self.get_current_population() == 1 {
             return Err(IxaError::IxaError(
                 "Cannot get a contact when there is only one person in the population.".to_string(),
             ));
         };
-        // get list of alive people
-        let alive_people = self.query_people((Alive, true));
+        // Get list of alive people (for now). May be expanded in the future to instead be
+        // list of alive people in the transmitter's contact setting or household.
+        // We will sample a random person from this list.
+        let mut alive_people = self.query_people((Alive, true));
         if alive_people.len() > 1 {
-            // get a random person from the list of alive people
-            // assign weights to each person in the list
+            // In the future, we might like to sample people from the list by weights according
+            // to some contact matrix.
             let mut weights = vec![1.0; alive_people.len()];
-            // get index of transmitee
-            let transmitee_index = alive_people
+            // Get the index of the transmitter.
+            let transmitter_index = alive_people
                 .iter()
-                .position(|&x| x == transmitee_id)
+                .position(|&x| x == transmitter_id)
                 .unwrap();
-            // set this weight equal to 0 so we don't select the transmitee as the contact
-            weights[transmitee_index] = 0.0;
-            let contact_id = self.sample_person_from_list(alive_people, &weights);
-            Ok(contact_id)
+            // Remove the transmitter from the list of contacts.
+            alive_people.remove(transmitter_index);
+            weights.remove(transmitter_index);
+            let contact_id = sample_person_from_list(self, &alive_people, &weights);
```

Comment:
Now you don't need weights, so why don't you just sample from the uniform distribution of (0..alive_people) and then floor, inline, rather than having a separate function.